assembl/graphql/landing_page.py

```diff
@@ -0,0 +1,137 @@
+                return self.__mapper__.primary_key_from_instance(self)[0]
+            return getattr(self, graphene_type._meta.id, None)
+
+
+class CreateLandingPageModule(graphene.Mutation):
+
+    class Input:
+        type_identifier = graphene.String()
+        enabled = graphene.Boolean()
+        order = graphene.Float()
+        configuration = graphene.String()
+
+    landing_page_module = graphene.Field(lambda: LandingPageModule)
+
+    @staticmethod
+    @abort_transaction_on_exception
+    def mutate(root, args, context, info):
+        cls = models.LandingPageModule
+        discussion_id = context.matchdict['discussion_id']
+        user_id = context.authenticated_userid or Everyone
+        permissions = get_permissions(user_id, discussion_id)
+        allowed = cls.user_can_cls(
+            user_id, CrudPermissions.CREATE, permissions)
+        if not allowed or (allowed == IF_OWNED and user_id == Everyone):
+            raise HTTPUnauthorized()
+        configuration = args.get('configuration')
+        order = args.get('order')
+        enabled = args.get('enabled')
+        module_type_identifier = args.get('type_identifier')
+        with cls.default_db.no_autoflush as db:
+            module_type = db.query(models.LandingPageModuleType).filter(
+                models.LandingPageModuleType.identifier == module_type_identifier).one()
+            saobj = cls(
+                discussion_id=discussion_id,
+                configuration=configuration,
+                order=order,
+                enabled=enabled,
+                module_type=module_type
+            )
+            db.add(saobj)
+            db.flush()
+
+        return CreateLandingPageModule(landing_page_module=saobj)
+
+
+class UpdateLandingPageModule(graphene.Mutation):
+
+    class Input:
+        id = graphene.ID(required=True)
+        enabled = graphene.Boolean()
+        order = graphene.Float()
+        configuration = graphene.String()
+
+    landing_page_module = graphene.Field(lambda: LandingPageModule)
+
+    @staticmethod
+    @abort_transaction_on_exception
+    def mutate(root, args, context, info):
+        cls = models.LandingPageModule
+        discussion_id = context.matchdict['discussion_id']
+        user_id = context.authenticated_userid or Everyone
+        configuration = args.get('configuration')
+        order = args.get('order')
+        enabled = args.get('enabled')
+        module_id = args.get('id')
+        module_id = int(Node.from_global_id(module_id)[1])
+        permissions = get_permissions(user_id, discussion_id)
+        allowed = cls.user_can_cls(
+            user_id, CrudPermissions.UPDATE, permissions)
+        if not allowed or (allowed == IF_OWNED and user_id == Everyone):
+            raise HTTPUnauthorized()
+
+        with cls.default_db.no_autoflush as db:
+            module = db.query(models.LandingPageModule).filter(
+                models.LandingPageModule.id == module_id).first()
```

Comment:
Prefer `one()` instead of `first()` here. `one()` raises `NoResultFound`, `first()` returns `None`

Reply:
done